server/routes/activities/create-an-activity/handlers/name.ts

```diff
@@ -19,10 +20,21 @@ export default class NameRoutes {
   }
 
   POST = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
     const { category } = req.session.createJourney
-    req.session.createJourney.name = req.body.name
+    req.session.createJourney.name = convertToTitleCase(req.body.name)
+
+    const activities = await this.activitiesService.getActivities(true, user)
+    const activityNames = activities.map(({ activityName }) => activityName)
+    const duplicateName = activityNames.find(activity => activity === req.session.createJourney.name)
+
+    if (req.params.mode === 'edit' && duplicateName) {
```

Comment:
I think this might be missing a check.

When editing you can in effect be comparing the currently edited activity to itself.  For example the user could just confirm without actually changing the activity name but it would look like a duplicate even though it is looking at the exact same activity.

So I think perhaps you may need to factor in the presence of an ID check as well?